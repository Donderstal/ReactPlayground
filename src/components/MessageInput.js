import * as React from 'react'
import {StyleSheet, TouchableOpacity, Text, TextInput, View} from 'react-native';

class MessageInput extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            input: 'hey'
        }
    }

    logger(input) {
        const msg = input
        this.props.handleClick(msg)
    }

    render() {
        return (
            <View style={{flex: .25}}>
            
                <TextInput
                style={{backgroundColor: 'white', width: '90%', marginLeft: 'auto',
                marginRight: 'auto', padding: '3%'}} placeholder='Type here and click the button...'
                onChangeText={(input) => this.setState({input:input})}
                ></TextInput>

                <TouchableOpacity 
                onPress={() => this.logger(this.state.input)}
                style={{...this.props.style, 
                    border: StyleSheet.hairlineWidth,
                    borderRadius: 5, 
                    width: '35%', height: '45%', 
                    marginBottom: '2,5%',
                    marginTop: '2,5%',
                    marginLeft: 'auto',
                    marginRight: 'auto',
                    padding: '3%'}}>   
                    <Text style={{marginLeft: 'auto',
                    marginRight: 'auto'}}>Click me!</Text>
                </TouchableOpacity>      

            </View>
        )
    }

}

export default MessageInput
