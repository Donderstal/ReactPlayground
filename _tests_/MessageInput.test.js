import React from 'react';
import renderer from 'react-test-renderer';


import MessageInput from '../src/components/MessageInput';


test('test if mInput renders', () => {
     const tree = renderer.create(<MessageInput />).toJSON();
      expect(tree).toMatchSnapshot();
});