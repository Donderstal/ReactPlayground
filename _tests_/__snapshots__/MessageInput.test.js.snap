// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if mInput renders 1`] = `
<View
  style={
    Object {
      "flex": 0.25,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Type here and click the button..."
    style={
      Object {
        "backgroundColor": "white",
        "marginLeft": "auto",
        "marginRight": "auto",
        "padding": "3%",
        "width": "90%",
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "border": 0.5,
        "borderRadius": 5,
        "height": "45%",
        "marginBottom": "2,5%",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "2,5%",
        "opacity": 1,
        "padding": "3%",
        "width": "35%",
      }
    }
  >
    <Text
      style={
        Object {
          "marginLeft": "auto",
          "marginRight": "auto",
        }
      }
    >
      Click me!
    </Text>
  </View>
</View>
`;
